controllers/spotifyController.js

```diff
@@ -0,0 +1,22 @@
+const source = require('rfr');
+
+const { getSpotifyAuthToken } = source('lib/spotify/spotifyApi');
+
+const SpotifyController = {
+    get(req, res) {
+        const userid = req.query.state;
+        const { code } = req.query;
+        let error = req.query.error || 'Missing the userid and the authentication code';
+
+        if (userid && code) {
+            getSpotifyAuthToken(userid, code);
+            error = null;
+        }
+
+        return res.render('authenticate', {
+            error,
```

Comment:
I suspect that this is actually buggy, but this was there before. This looks like it will always display an error message even if it was successful.

Reply:
I believe authenticate.handlebars checks if error is null, then displays either success or failure

Reply:
oh ya. I missed that  if statement